src/core/load-fragment.c

```diff
@@ -1416,6 +1416,98 @@ int config_parse_exec_cpu_sched_prio(const char *unit,
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_(mount_options_free_allp) MountOptions *options = NULL;
+        ExecContext *c = data;
+        const char *p = rvalue;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue)) {
+                c->root_image_options = mount_options_free_all(c->root_image_options);
+                return 0;
+        }
+
+        for (;;) {
+                _cleanup_free_ char *partition = NULL, *mount_options = NULL, *tuple = NULL;
+                MountOptions *o = NULL;
+                unsigned int partition_number = 0;
+
+                r = extract_first_word(&p, &tuple, WHITESPACE, EXTRACT_UNQUOTE);
+                if (r == 0)
+                        break;
+                if (r == -ENOMEM)
+                        return log_oom();
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse %s: %s", lvalue, rvalue);
+                        return 0;
+                }
+
+                r = extract_first_word((const char **)&tuple, &partition, ":", EXTRACT_UNQUOTE|EXTRACT_DONT_COALESCE_SEPARATORS);
+                if (r == 0)
+                        continue;
+                if (r == -ENOMEM)
+                        return log_oom();
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse %s: %s", lvalue, rvalue);
+                        return 0;
+                }
+
+                if (tuple && tuple[-1] == ':') {
+                        r = extract_first_word((const char **)&tuple, &mount_options, NULL, EXTRACT_UNQUOTE|EXTRACT_DONT_COALESCE_SEPARATORS);
+                        if (r == -ENOMEM)
+                                return log_oom();
+                        if (r < 0) {
+                                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse %s: %s", lvalue, rvalue);
+                                return 0;
+                        }
+                }
+
+                /* Format is either '0:foo' or 'foo' (0 is implied) */
+                if (isempty(mount_options))
+                        free_and_replace(mount_options, partition);
+                else {
+                        r = safe_atou(partition, &partition_number);
+                        if (r < 0) {
+                                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse partition number from \"%s\", ignoring: %m", partition);
+                                continue;
+                        }
+                }
```

Comment:
i think we should do specifier expansion on the option string

Reply:
@poettering sorry, which specifiers? You mean like in this commit: https://github.com/systemd/systemd/commit/e031166e15184d861bc ?

Reply:
Did you mean to pass the string through unit_full_printf by any chance?

Reply:
Changed to pass the string through unit_full_printf, and added a test to make sure %%foo becomes %foo when querying via DBUS.